src/Raven.Client/Documents/Session/SessionTimeSeriesBase.cs

```diff
@@ -122,6 +122,48 @@ public void Delete(DateTime? from = null, DateTime? to = null)
             {
                 Session.Defer(new TimeSeriesBatchCommandData(DocId, Name, appends: null, deletes: new List<TimeSeriesOperation.DeleteOperation> { op }));
             }
+
+            RemoveFromCacheIfNeeded(from, to);
+        }
+
+        private void RemoveFromCacheIfNeeded(DateTime? from = null, DateTime? to = null)
```

Comment:
Maybe we can simply it to sth like:

```csharp
        private void RemoveFromCacheIfNeeded(DateTime? from = null, DateTime? to = null)
        {
            if (Session.TimeSeriesByDocId.TryGetValue(DocId, out var cache) == false)
                return;

            if (from == null && to == null)
            {
                cache.Remove(Name);
                return;
            }

            if (cache.TryGetValue(Name, out var ranges) && ranges.Count > 0)
            {
                from ??= DateTime.MinValue;
                to ??= DateTime.MaxValue;

                ranges.RemoveAll(range => range.From <= from && range.To >= to);
            }
        }
```

What do you think?

Reply:
Great!! done :)